src/os/bluestore/BlueStore.cc

```diff
@@ -7213,8 +7213,7 @@ int BlueStore::_collection_list(
   if (!pnext)
     pnext = &static_next;
 
-  if (start == ghobject_t::get_max() ||
-    start.hobj.is_max()) {
+  if (start.is_max()) {
```

Comment:
this is important because we want to skip out even if ghobject_t::max is false but hobj.max is true.